File: Transpiler/test cases/IF_nested.cpp
/* 
IF nested 
valid
*/
#include <iostream>
#include <fstream>
using namespace std;

int main () {

int x = 1;
int y = 2;
int z = 3;
bool b = true;
float q = 1.1;
float w = 2.2;

if (x < y)
{ 
    cout << "x < y is true" << endl;
    if (y < z)
    {
        cout << "y < z is true, incrementing z" << endl;
        z++;
    }
    else if (z < 4)
    {
        cout << "z < 4 is true, setting z = x" << endl;
        z=x;
    }
}
else 
{
    cout << "x < y is false, incrementing y" << endl;
    y++;
}

cout << "After if statements: x=" << x << ", y=" << y << ", z=" << z << endl;

return 0;
}

File: Transpiler/test cases/IF_nested.py
def main():
    x = 1
    y = 2
    z = 3
    b = True
    q = 1.1
    w = 2.2
    if x < y:
        print("x < y is true")
        if y < z:
            print("y < z is true, incrementing z")
            z = z + 1
        elif z < 4:
            print("z < 4 is true, setting z = x")
            z = x
    else:
        print("x < y is false, incrementing y")
        y = y + 1
    print("After if statements: x=" + str(x) + ", y=" + str(y) + ", z=" + str(z))
    return 0

if __name__ == "__main__":
        main()
